benchmark/single-source/SetTests.swift

```diff
@@ -49,12 +49,50 @@ let setY = Set(half ..< size)         // 200 ..< 400
 // Two sets with 100% overlap, but different bucket counts (let's not make it
 // too easy...)
 let setP = Set(0 ..< size)
-let setQ: Set<Int> = {
-  var set = Set(0 ..< size)
-  set.reserveCapacity(2 * size)
-  return set
+let setQ = Set(0 ..< size)
+
+// Construction of empty array.
+let arrayE: Array<Int> = []
+let arrayOE: Array<Box<Int>> = []
+
+// Construction kit for arrays with 25% overlap
+let arrayAB = Array(0 ..< size)                              //   0 ..< 400
+let arrayCD = Array(size ..< 2 * size)                       // 400 ..< 800
+let arrayBC = Array(size - quarter ..< 2 * size - quarter)   // 300 ..< 700
+let arrayB = Array(size - quarter ..< size)                  // 300 ..< 400
+
+let arrayOAB = arrayAB.map(Box.init)
+let arrayOCD = arrayCD.map(Box.init)
+let arrayOBC = arrayBC.map(Box.init)
+let arrayOB = arrayB.map(Box.init)
+
+// Construction kit for arrays with 50% overlap
+let arrayXY = Array(0 ..< size)           //   0 ..< 400
+let arrayYZ = Array(half ..< size + half) // 200 ..< 600
+let arrayY = Array(half ..< size)         // 200 ..< 400
+
+// Two arrays with 100% overlap, but different bucket counts (let's not make it
+// too easy...)
+let arrayP = Array(0 ..< size)
+let arrayQ: Array<Int> = {
```

Comment:
As I [said before](https://github.com/apple/swift/pull/24156#discussion_r280842909), we don't need a shuffled `arrayQ` at all. All `Int100` comparisons between a `Set` and a `Sequence` can be performed on `setP` and `arrayP` only. I'll point out all the places below.